src/light_pcapng_ext.c

```diff
@@ -445,9 +445,48 @@ uint8_t get_timestamp_resolution_precision(uint64_t timestamp_resolution) {
 	return precision;
 }
 
+
+int light_write_interface_block(light_pcapng pcapng, const light_packet_interface * packet_interface)

```

Comment:
most of this method content duplicates what's in light_write_packet, could you de-duplicate it?

Reply:
I took out the duplication, but it was there because I didn't understand why +1 was being added to the lengths for the description. I had removed it in my function, but I didn't want to risk causing a bug. 

Why is +1 added to the option lengths?

Reply:
The size of the memory that needs to be copied from the string is `strlen(some_string) + 1` the 1 is for the zero termination character.